language/move-binary-format/src/file_format.rs

```diff
@@ -647,23 +677,24 @@ impl AbilitySet {
     }
 
     /// For a polymorphic type, its actual abilities correspond to its declared abilities but
-    /// predicated on its type arguments having that ability. For `Key`, instead of needing
-    /// the same ability, the type arguments need `Store`
+    /// predicated on its non-phantom type arguments having that ability. For `Key`, instead of needing
+    /// the same ability, the type arguments need `Store`.
     pub fn polymorphic_abilities(
         declared_abilities: Self,
-        type_argument_abilities: impl IntoIterator<Item = Self>,
+        type_arguments: impl IntoIterator<Item = (Self, bool)>,
     ) -> Self {
         // Conceptually this is performing the following operation:
         // For any ability 'a' in `declared_abilities`
         // 'a' is in the result only if
-        //   for all 'ti' in `type_argument_abilities`, a.required() is a subset of ti
+        //   for all (abi_i, is_phantom_i) in `type_arguments` s.t. !is_phantom then a.required() is a subset of abi_i
```

Comment:
mega nit:, to me abi meands abilities :P maybe just write it out, or ignore me. Either way :)